gengokit/middlewares/middlewares.go

```diff
@@ -0,0 +1,63 @@
+// Package middlewares renders the service and endpointe middleware files in NAME-service/middlewares/.
+package middlewares
+
+import (
+	"io"
+
+	"github.com/pkg/errors"
+
+	"github.com/TuneLab/go-truss/gengokit"
+	"github.com/TuneLab/go-truss/gengokit/middlewares/templates"
+)
+
+// EndpointsPath is the path to the Endpoints middleware file that package middlewares renders
+const EndpointsPath = "NAME-service/middlewares/endpoints.gotemplate"
+
+// ServicePath is the path to the Service middleware file that package middlewares renders
+const ServicePath = "NAME-service/middlewares/service.gotemplate"
+
+// New returns a Middleware which can render EndpointsFile and ServiceFile as
+// well as read in previous versions of each respective file
+func New() *Middlewares {
```

Comment:
So I am not attached to this method though here is my reasoning. Please tell me what you think.

I added this method to be symetrical with the `handler` package. Though at this point, it does not take any parameters, does no setup, and cannot fail. This is because the middlewares are not doing anything complicated.

But new features may add requirements for setting things up. (e.g. updating middlewares based on the .proto file, inserting middlewares). So doing a `new(middlewares.Middlewares)` and then just calling `Load...` or `Render` may be a problem. 

Though this may be a case of wanting to add a wiggle when I don't need it.

If you think my reasoning is not up to snuff, I can change it to `new(middlewares.Middlewares)` and update it later to a `New` method if setup is required.